doc/basics.md

```diff
@@ -77,8 +77,7 @@ Once you have an element, you can navigate it with idiomatic C++ iterators, oper
 
 * **Extracting Values:** You can cast a JSON element to a native type: `double(element)` or
   `double x = json_element`. This works for double, uint64_t, int64_t, bool,
-  dom::object and dom::array. An exception is thrown if the cast is not possible. You can also use is<*typename*>() to test if it is a
-  given type, or use the `type()` method: e.g., `element.type() == dom::element_type::DOUBLE`. Instead of casting, you can use get<*typename*>() to get the value: casts and get<*typename*>() can be used interchangeably. You can use a variant usage of get<*typename*>() with error codes to avoid exceptions: e.g.,  
+  dom::object and dom::array. An exception is thrown if the cast is not possible. You can also use is<*typename*>() to test if it is a given type, or use the equivalent without templates (`is_uint64()`, `is_uint64()`, `is_number()`, `is_double()`, , `is_bool()`, `is_integer()`, `is_string()`, `is_array()`, `is_object()`) or use the `type()` method: e.g., `element.type() == dom::element_type::DOUBLE`. Instead of casting, you can use get<*typename*>() to get the value: casts and get<*typename*>() can be used interchangeably. You can also use equivalent functions without templates: e.g., `get_string()`, `get_int64()`, `get_uint64()`, `get_double()` and `get_bool()`. You can use a variant usage of get<*typename*>() with error codes to avoid exceptions: e.g.,
```

Comment:
In "Instead of casting," I'd probably say "If you use error codes instead of exceptions, you can cast with get ...." and maybe have it in a new paragraph. Reason: helps users limit the choices they have to make by telling them which to use when.

Reply:
@lemire backing up a tiny bit, I think I'd personally rather we not advertise these methods heavily at all--just leave them in the API docs. IMO, the two preferred methods of casting we signal should be:

* Exceptions: `double d = x` or `double(x)`
* Error codes: `double d; error = x.get(d);

The others may have niche cases, but putting them out there will bring too much confusion on what would otherwise be a strong, clear message.

Reply:
(I can mess with it after checkin too though. There's nothing *wrong* with what you have here.)

Reply:
I have done so. Two paragraphs. One with exceptions, one without. The other functions have been removed.